Library/ViewModels/PledgeCTAContainerViewViewModel.swift

```diff
@@ -0,0 +1,92 @@
+import KsApi
+import Prelude
+import ReactiveExtensions
+import ReactiveSwift
+
+public protocol PledgeCTAContainerViewViewModelInputs {
+  func configureWith(project: Project, user: User)
+}
+
+public protocol PledgeCTAContainerViewViewModelOutputs {
+  var buttonBackgroundColor: Signal<UIColor, Never> { get }
+  var buttonTitleText: Signal<String, Never> { get }
+  var rewardTitle: Signal<String, Never> { get }
+  var stackViewIsHidden: Signal<Bool, Never> { get }
+}
+
+public protocol PledgeCTAContainerViewViewModelType {
+  var inputs: PledgeCTAContainerViewViewModelInputs { get }
+  var outputs: PledgeCTAContainerViewViewModelOutputs { get }
+}
+
+public final class PledgeCTAContainerViewViewModel: PledgeCTAContainerViewViewModelType,
+  PledgeCTAContainerViewViewModelInputs, PledgeCTAContainerViewViewModelOutputs {
+  public init() {
+    let projectAndUser = self.projectAndUserProperty.signal.skipNil()
+    let project = projectAndUser.map { $0.0 }
+
+    let pledgeState = project
+      .map { pledgeStateButton(project: $0) }
+
+    let backingEvent = projectAndUser
+      .switchMap { project, user in
+        AppEnvironment.current.apiService.fetchBacking(forProject: project, forUser: user)
+          .materialize()
+    }
+    let backing = backingEvent.values()
+    let projectAndBacking = Signal.combineLatest(project, backing)
+
+    self.buttonTitleText = pledgeState.map { $0.buttonTitle }
+    self.buttonBackgroundColor = pledgeState.map { $0.buttonBackgroundColor }
+    self.stackViewIsHidden = pledgeState.map { $0.stackViewIsHidden }
+
+    self.rewardTitle = projectAndBacking
+      .map { (arg) -> String in
+
+        let (project, backing) = arg
+        let basicPledge = formattedAmount(for: backing)
+        let amount = Format.formattedCurrency(
+          basicPledge,
+          country: project.country,
+          omitCurrencyCode: project.stats.omitUSCurrencyCode
+        )
+        guard let rewardTitle = backing.reward?.title else { return "\(amount)" }
+        return "\(amount) • \(rewardTitle)"
+      }
+  }
+
+  fileprivate let projectAndUserProperty = MutableProperty<(Project, User)?>(nil)
+  public func configureWith(project: Project, user: User) {
+    self.projectAndUserProperty.value = (project, user)
+  }
+
+  public var inputs: PledgeCTAContainerViewViewModelInputs { return self }
+  public var outputs: PledgeCTAContainerViewViewModelOutputs { return self }
+
+  public let buttonBackgroundColor: Signal<UIColor, Never>
+  public let buttonTitleText: Signal<String, Never>
+  public let rewardTitle: Signal<String, Never>
+  public let stackViewIsHidden: Signal<Bool, Never>
+}
+
+private func pledgeStateButton(project: Project) -> PledgeStateCTAType {
```

Comment:
Can we separate this section?

```suggestion

// MARK: - Functions

private func pledgeStateButton(project: Project) -> PledgeStateCTAType {
```